src/agent/application.hpp

```diff
@@ -109,15 +109,23 @@ class Application : private NonCopyable
     /**
      * This method initializes the Application instance.
      *
+     * @param[in] aThreadEnabled  A boolean indicates whether Thread is enabled or not
      */
-    void Init(void);
+    void Init(bool aThreadEnabled);
 
     /**
      * This method de-initializes the Application instance.
      *
      */
     void Deinit(void);
 
+    /**
+     * Called when the Thread enabled state is changed.
+     *
+     * @param[in] aThreadEnabled  A boolean indicates whether Thread is enabled or not
+     */
+    void OnThreadEnabledChanged(bool aThreadEnabled);
```

Comment:
`OnThreadEnabledChanged()` is named as if signaling an event with `aThreadEnabled` input. 
- The  `aThreadEnabled` name itself may a bit confusing.  
- E.g., we have `otThreadSetEnabled()` which I think is not related to this.

If I understand the intention, the idea here is for next layer code (e.g. Android) to use this after it has set vendor/product name so we can start advertising "meshcop" service in AIL (i.e. we are ready to enable the "meshcop" agent).

How about renaming this to make it clear? Maybe `SetMeshcopAgentEnabled()`?

Reply:
Yeah, the Android level Thread enabled state is a bit different with `otThreadSetEnabled()`: When Thread is disabled in Android, apps are not allowed to call majority of the Android Thread APIs (i.e. apps are returned error code ERROR_THREAD_DISABLED), while some of the APIs which don't operate over radio are still allowed.